packages/ccd-js-gen/src/lib.ts

```diff
@@ -894,23 +1041,59 @@ function generateContractEntrypointCode(
                     },
                 ],
                 returnType: 'SDK.Parameter.Type',
-            })
-            .setBodyText(
-                [
-                    ...receiveParameter.code,
-                    `return ${receiveParameter.id};`,
-                ].join('\n')
-            );
+                statements: [
+                    `return SDK.Parameter.fromBase64SchemaType(${base64ReceiveParameterSchemaTypeId}, ${createReceiveParameterJsonFnId}(${parameterId}));`,
+                ],
+            });
+
+            contractSourceFile.addFunction({
+                docs: [
+                    [
+                        `Construct WebWallet parameter type used in update transaction for '${entrypointName}' entrypoint of the '${contractName}' contract.`,
+                        `@param {${receiveParameterTypeId}} ${parameterId} The structured parameter to construct from.`,
+                        '@returns The smart contract parameter support by the WebWallet.',
+                    ].join('\n'),
+                ],
+                isExported: true,
+                name: createReceiveParameterFnWebWalletId,
+                parameters: [
+                    {
+                        type: receiveParameterTypeId,
+                        name: parameterId,
+                    },
+                ],
+                statements: [
+                    'return {',
+                    `    parameters: ${createReceiveParameterJsonFnId}(${parameterId}),`,
+                    '    schema: {',
+                    "        type: 'TypeSchema' as const,",
```

Comment:
Do we need the `const` here?

```suggestion
                    "        type: 'TypeSchema' ",
```

Reply:
This is needed to avoid typescript inferring `type: 'TypeSchema'` to `string`, which will result in type errors when this is passed as `TypedSmartContractParameters`.
Having `as const` makes it the type `'TypeSchema'`.